### PR TITLE
Move plugin to next-translate-plugin package

### PR DESCRIPTION
**Pull request by aralroca**
@StLyn4 as a major contributor to version 2, I would appreciate it if you could give me some feedback on this change 😊

-----

## About this PR:

In version 2 of the **next-translate** library, we have made a significant change to the way the Webpack plugin is implemented. This change improves the library by solving many of the issues reported by users and makes it more powerful. However, it also means that two of the library's rules have been broken - it is now slightly larger and has a dependency on a specific version of the TypeScript parser.

To address this, we have split the library into two separate packages. The main package, **[next-translate](https://github.com/aralroca/next-translate)**, is used as before and handles all of the translation functionality. The new package, **[next-translate-plugin](https://github.com/aralroca/next-translate-plugin)**, is used specifically for the Webpack plugin and should only be used in the `next.config.js` file.

It's worth noting that by splitting the library into two packages and only using the `next-translate-plugin` in the `next.config.js` file, we are able to ensure that the extra kilobytes added by the TypeScript parser are only loaded during the build process and not included in the final bundle that is served to the client. This helps to keep the size of the final bundle as small as possible and improves the performance of your application. Additionally, by listing the `next-translate-plugin` will be installed as devDependency to reduce the extra kilobytes in the pipeline.

To migrate your project to use version 2, you will need to make the following changes:

- Install the **next-translate-plugin** package as a devDependency using `yarn add -D next-translate-plugin` or `npm install --save-dev next-translate-plugin`
- In your `next.config.js` file, replace `require('next-translate')` with `require('next-translate-plugin')`

Example of `next.config.js`:

```js
const nextTranslate = require('next-translate-plugin')

module.exports = nextTranslate()
```

### CODE REVIEW

File: examples/complex/i18n.js [hunk 1]
```diff
@@ -13,6 +13,6 @@ module.exports = {
     prefix: '${',
     suffix: '}',
   },
-  loadLocaleFrom: (locale, namespace) =>
-    import(`./src/translations/${namespace}_${locale}`).then((m) => m.default),
+  loadLocaleFrom: async (locale, namespace) =>
+    require(`./src/translations/${namespace}_${locale}`),
```
Comment (aralroca) [inline examples/complex/i18n.js:17]: For this we need to find a better way before 2.0 to fix this issue: https://github.com/aralroca/next-translate/issues/851

**Comment by StLyn4** [comment]
Calling tsc this way does not work on Windows - it breaks the build process (the change was in commit [f134a17da01e40bb391342827bc04b9e1afc98b6](https://github.com/aralroca/next-translate/commit/f134a17da01e40bb391342827bc04b9e1afc98b6), not in this PR. The old version works):
`"tsc": "./node_modules/typescript/bin/tsc -p tsconfig.json && ./node_modules/typescript/bin/tsc -p tsconfig-cjs.json && node build-packages.js"`

We also need to add changes to the README of the main package. There are still old installation instructions.

The rest LGTM and works fine if fix the "tsc" script.
By the way, as far as I remember, when I did the first tests in the fall, the bundle analyzer told me that the size of the final assembly had not changed in any way after switching to typescript. As I understand it, tree shaking works. But, nevertheless, I don’t see anything wrong with moving the plugin into a separate module 😊

**Comment by aralroca** [comment]
Thanks @StLyn4 . I changed to `tsc` again.

**Comment by BjoernRave** [comment]
but isn't next.js moving to swc and trying to deprecate webpack in the long term? Would this path mean that we can not benefit of the advancements on the swc front?

**Comment by StLyn4** [comment]
> but isn't next.js moving to swc and trying to deprecate webpack in the long term? Would this path mean that we can not benefit of the advancements on the swc front?

It's possible that I misunderstood the messages, but the plugin has nothing to do with SWC. I mean SWC replaced babel, not webpack. In turn, Turbopack is going to replace Webpack.
But, we don't currently have the ability to create plugins for it (at least for now), so there's nothing we can do about it.
Besides, these are just my guesses, but I think that plugins will have to be done in Rust, so the plugin will have to be rewritten one way or another in such a situation. In this case, separating into separate modules will play a good role, as 2 independent modules can be created for Webpack and Turbopack. Maybe even something more exotic. While the main module can exist and be updated independently 😄

**Comment by aralroca** [comment]
> but isn't next.js moving to swc and trying to deprecate webpack in the long term? Would this path mean that we can not benefit of the advancements on the swc front?

We are enhancing the current plugin, instead of regex now we are parsing the files correctly. Currently, Turbopack is in alpha stage and has room for improvement, and it is likely that it will support Webpack plugins. If not, we are going to migrate in the future when Turbopack support it.

**Comment by BjoernRave** [comment]
thanks for clearing that up, I was mixing things up there :) thanks alot to you for your contributions to this library

**Pull request merged**